public_html/app/Controllers/Users.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace VFramework\Controllers;
+
+use VFramework\Helpers\UrlHelper;
+use VFramework\Libraries\Controller;
+use VFramework\Models\User;
+use VFramework\Tools\Request;
+use VFramework\Tools\Validator;
+
+
+class Users extends Controller
+{
+
+    /**
+     * @var mixed
+     */
+    private $model;
+    /**
+     * @var Validator
+     */
+    public $validator;
+    /**
+     * @var Request
+     */
+    private $request;
+
+    public const LOGIN_RULES = [
+        'email' => ['required', 'exists'],
+        'password' => ['required']
+    ];
+
+    public const REGISTRATION_RULES = [
+        'email' => ['required', 'emailIsUnique'],
+        'password' => ['required'],
+        'confirm_password' => ['passwordsMatch']
+    ];
+
+    /**
+     * Users constructor.
+     * @param Validator $validator
+     */
+    public function __construct(Validator $validator)
+    {
+        $this->validator = $validator;
+        $this->model = new User();
+        $this->request = new Request();
+    }
+
+    public function index()
+    {
+        return $this->view('users/register');
+    }
+
+    // REGISTER
+    /**
+     * @param array $data
+     */
+    public function add()
+    {
+        // IF POST METHOD
+        if ($this->request->requested('POST')) {
+            // Sanitize POST data
+            $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+            // Init data
+            $data = [
```

Comment:
I think you can pass $_POST to validation and validate directory from $_POST . And then create function validated() which will return you validated parameters.